src/Form/matching.js

```diff
@@ -754,18 +830,29 @@ function getInputSubtype (input) {
     return getSubtypeFromType(type)
 }
 
+/**
+ * Retrieves the input variant
+ * @param {HTMLInputElement|Element} input
+ * @returns {SupportedVariants | ''}
+ */
+function getInputVariant (input) {
+    const type = getInputType(input)
+    return getVariantFromType(type)
+}
+
 /**
  * Remove whitespace of more than 2 in a row and trim the string
  * @param {string | null} string
  * @return {string}
  */
 const removeExcessWhitespace = (string = '') => {
+    string = string?.trim() || ''
```

Comment:
Just moved the trimming here so the length cutoff is more accurate. Form test added.